sympy/matrices/expressions/tests/test_matpow.py

```diff
@@ -24,7 +24,7 @@ def test_as_explicit_symbol():
     X = MatrixSymbol('X', 2, 2)
     assert MatPow(X, 0).as_explicit() == ImmutableMatrix(Identity(2))
     assert MatPow(X, 1).as_explicit() == X.as_explicit()
-    assert MatPow(X, 2).as_explicit() == (X.as_explicit())**2
+    assert MatPow(X, 2).as_explicit() == ((X.as_explicit())**2).expand()
```

Comment:
Why aren't these the same?

Reply:
Different evaluation paths, `MatPow(X, 2).as_explicit()` does not use actual matrix multiplication but rather builds up its own matrix as opposed to `X.as_explicit()` which first builds the base matrix then applying the optimizing multiplication on it.